### ISSUE TITLE
Disallow all block types from being embedded, until specifically supported.

### ISSUE THREAD

**Issue by stpaultim** [backdrop-contrib/ckeditor_blocks#22]
I just spun up a https://simplo.site/template/farmers-market-1 with this module to see how it works in an existing site with a bunch of modules already turned on and available. 

A bunch of "Block Types" provided by other modules show up. These are not specific blocks that are ready to be placed, but types of blocks that can be generated by other modules. 

We need to figure out how to filter all of these out. Current code is filtering out some known block types. 

Maybe the correct solution is to only allow legitimate blocks, not filter out the ones that should not show up.

Example: Fanciblock icons

---

## Core Block Types for MVP:
* Block: Reusable custom blocks - allow 
* Views: Block displays - allow (**ignoring block instance settings**)

## Core Block Types (nice to have):
* Book: navigation - allow?
* Contact: sitewide - allow?
* Menu: custom - allow?
* System: Menu - allow?
* Views: Block displays - allow (**support block instance settings**)

## Other Core Block Types (not likely used in content):
* Comment: recent comments - disallow (this should really be a view anyway)
* Dashboard  - disallow
* Field - disallow
* Layout: custom blocks - disallow
* Layout: hero blocks - disallow
* Locale: language switcher - disallow
* Node: syndicate - disallow
* Node: recent - disallow (should really be a view)
* Node: existing content - disallow
* Search - disallow
* System: main - disallow
* System: header - disallow
* System: breadcrumb - disallow
* System: powered-by - disallow
* System: page components - disallow
* User: login block - disallow
* Views: Exposed filter blocks - disallow

**Comment by stpaultim** [comment]
I've been told that "Blocks" are in a sense "Block Types" and that for the most part, this module is interested in Block Instances. However, we need to think more about how we can distinguish them easily for the purposes of this module. Backdrop does not track this difference directly, there there are characteristics that we could use as a filter. But, we need to research this some more.

**Comment by jenlampton** [comment]
Backdrop does know which Class is used to create a block instance, and that can be used to determine a "Block type". 

By default, it would be best if all "Block types" were specifically disabled (and not allowed to be embedded). We can go through each of the "block types" as they arise, test them, and enable them once we have confirmed they work, or after we have made whatever changes are necessary to get them working.

A good MVP would be to have some reasonable core block types supported. (I'll provide a list in the original issue description)
